### PR TITLE
Diameter of a Binary Tree in Java

### PR DESCRIPTION
**Pull request by ankitasantape**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

The diameter of a binary tree is the length of the longest path between any two nodes in a tree. This path may or may not pass through the root.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

DWoC
issue Addressed -  #1802

### CODE REVIEW

File: Java/Algorithms/NthFibonacciElement.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import java.util.Scanner;
+
+public class NthFoboElement {
```
Comment (atarax665) [inline Java/Algorithms/NthFibonacciElement.java]:
```suggestion
public class NthFibonacciElement {
```

File: Java/ds/Diameter_Of_A_BinaryTree.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		
+		Pair<Integer, Integer> output = new Pair<>();
+		output.height = height;
+		output.diameter = diameter;
+		return output;
+	}
+
+	public static void main(String[] args) {
+		BinaryTreeNode<Integer> root = new BinaryTreeNode<>(1);
+		root.left = new BinaryTreeNode<>(2);
```
Comment (atarax665) [inline Java/ds/Diameter_Of_A_BinaryTree.java]: Dynamic inputs are preferred.

File: Java/ds/Diameter_Of_A_BinaryTree.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		root.right = new BinaryTreeNode<>(3);
+		root.left.left = new BinaryTreeNode<>(4);
+		root.left.right = new BinaryTreeNode<>(5);
+		root.left.left.left = new BinaryTreeNode<>(6);
+		root.left.left.right = new BinaryTreeNode<>(7);
+		
+		System.out.println( heightDiameter(root).diameter);
+	}
+
+
```
Comment (atarax665) [inline Java/ds/Diameter_Of_A_BinaryTree.java]: Remove these extra blank spaces.

File: Java/ds/Diameter_Of_A_BinaryTree.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+		root.left.left = new BinaryTreeNode<>(4);
+		root.left.right = new BinaryTreeNode<>(5);
+		root.left.left.left = new BinaryTreeNode<>(6);
+		root.left.left.right = new BinaryTreeNode<>(7);
+		
+		System.out.println( heightDiameter(root).diameter);
+	}
+
+	
+}
```
Comment (atarax665) [inline Java/ds/Diameter_Of_A_BinaryTree.java:109]: Add sample input/output in commented form.

File: Java/Algorithms/NthFibonacciElement.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    
+    public static int fibonacciNumber(int n)
+    {
+
```
Comment (atarax665) [inline Java/Algorithms/NthFibonacciElement.java]: Remove this line.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Review by iamrajiv — APPROVED** [review]

**Comment by HarshCasper** [comment]
@ankitasantape 

Please remove the Merge Conflicts 👍

**Pull request merged**